lms/services/roster.py

```diff
@@ -316,6 +326,127 @@ def fetch_canvas_group_roster(self, canvas_group: LMSSegment) -> None:
             update_columns=["active", "updated"],
         )
 
+    def fetch_canvas_sections_roster(self, lms_course: LMSCourse) -> None:
```

Comment:
Here we go...

This is a long method, the main complexity here is matching the API and LTI worlds. It's not complex code but there's plenty of steps.